pkg/artifactory/repositories.go

```diff
@@ -68,29 +71,31 @@ type ContentSynchronisationSource struct {
 }
 
 type RemoteRepositoryBaseParams struct {
-	Key                      string `hcl:"key" json:"key,omitempty"`
-	Rclass                   string `json:"rclass"`
-	PackageType              string `hcl:"package_type" json:"packageType,omitempty"`
-	Url                      string `hcl:"url" json:"url"`
-	Username                 string `hcl:"username" json:"username,omitempty"`
-	Password                 string `hcl:"password" json:"password,omitempty"`
-	Proxy                    string `hcl:"proxy" json:"proxy"`
-	Description              string `hcl:"description" json:"description,omitempty"`
-	Notes                    string `hcl:"notes" json:"notes,omitempty"`
-	IncludesPattern          string `hcl:"includes_pattern" json:"includesPattern,omitempty"`
-	ExcludesPattern          string `hcl:"excludes_pattern" json:"excludesPattern,omitempty"`
-	RepoLayoutRef            string `hcl:"repo_layout_ref" json:"repoLayoutRef,omitempty"`
-	RemoteRepoLayoutRef      string `json:"remoteRepoLayoutRef"`
-	HardFail                 *bool  `hcl:"hard_fail" json:"hardFail,omitempty"`
-	Offline                  *bool  `hcl:"offline" json:"offline,omitempty"`
-	BlackedOut               *bool  `hcl:"blacked_out" json:"blackedOut,omitempty"`
-	XrayIndex                *bool  `hcl:"xray_index" json:"xrayIndex,omitempty"`
-	PropagateQueryParams     bool   `hcl:"propagate_query_params" json:"propagateQueryParams"`
-	PriorityResolution       bool   `hcl:"priority_resolution" json:"priorityResolution"`
-	StoreArtifactsLocally    *bool  `hcl:"store_artifacts_locally" json:"storeArtifactsLocally,omitempty"`
-	SocketTimeoutMillis      int    `hcl:"socket_timeout_millis" json:"socketTimeoutMillis,omitempty"`
-	LocalAddress             string `hcl:"local_address" json:"localAddress,omitempty"`
-	RetrievalCachePeriodSecs int    `hcl:"retrieval_cache_period_seconds" json:"retrievalCachePeriodSecs,omitempty"`
+	Key                      string   `hcl:"key" json:"key,omitempty"`
+	ProjectKey               string   `json:"projectKey"`
```

Comment:
Here you don't define an hcl, but below you do - I just want to make sure you understand how this will work

Reply:
I keep thinking it derives the hcl attribute from the json, but it doesn't obviously. Doh!